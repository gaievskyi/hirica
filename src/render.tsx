import { Suspense } from "react";

import { render } from "@testing-library/react";
import { SessionProvider } from "next-auth/react";

import type { RenderOptions } from "@testing-library/react";
import type { Session } from "next-auth";
import type { FC, PropsWithChildren, ReactElement } from "react";

type Props = PropsWithChildren & unknown;

const mockSession: Session = {
  expires: `${Date.now() * 100}`,
};

const Providers: FC<Props> = ({ children }) => (
  <Suspense>
    <SessionProvider session={mockSession}>{children}</SessionProvider>
  </Suspense>
);

const mockRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: Providers, ...options });

export * from "@testing-library/react";
export { mockRender as render };
