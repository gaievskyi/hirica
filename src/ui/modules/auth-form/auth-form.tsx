import Link from "next/link";

type AuthFormProps = {
  action: () => void;
};

export const AuthForm: React.FC<AuthFormProps> = ({ action }) => (
  <form className="flex flex-col gap-10">
    <Link href="/" className="text-2xl font-semibold">
      Hirica â€¢ Hire and find job anonymously
    </Link>
    <div className="flex flex-col gap-2">
      <label htmlFor="email">E-mail</label>
      <input
        required
        type="email"
        name="email"
        className="rounded-full border-2 border-black p-4"
        placeholder="john.doe@email.com"
      />
    </div>
    <div className="flex items-center justify-center gap-3">
      <button
        onClick={action}
        type="submit"
        className="rounded-full bg-black p-4 text-white hover:bg-black/80 "
      >
        Next
      </button>
      <span>or continue with</span>
      <button
        type="submit"
        className="rounded-full bg-black p-4 text-white hover:bg-black/80"
      >
        Web 3.0
      </button>
      <button
        type="submit"
        className="rounded-full bg-black p-4 text-white hover:bg-black/80"
      >
        LinkedIn
      </button>
      <button
        type="submit"
        className="rounded-full bg-black p-4 text-white hover:bg-black/80"
      >
        GitHub
      </button>
    </div>
  </form>
);
