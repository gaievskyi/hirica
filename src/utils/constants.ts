import { env } from "~/env.mjs"

export const IS_DEVELOPMENT = env.NODE_ENV === "development"
export const IS_TEST = env.NODE_ENV === "test"
export const IS_PRODUCTION = env.NODE_ENV === "production"
