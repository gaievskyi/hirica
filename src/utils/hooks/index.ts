export * from './use-interception-observer'
