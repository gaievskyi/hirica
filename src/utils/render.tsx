import Router from 'next/router'
import { Suspense } from 'react'

import { render } from '@testing-library/react'
import { SessionProvider } from 'next-auth/react'
import { RouterContext } from 'next/dist/shared/lib/router-context'

import type { RenderOptions } from '@testing-library/react'
import type { Session } from 'next-auth'
import type { FC, PropsWithChildren, ReactElement } from 'react'

import { createMockRouter } from './test-utils'

type MockProps = PropsWithChildren & unknown

const mockSession: Session = {
  expires: `${Date.now() * 100}`,
}

const mockRouter = createMockRouter(Router)

const MockApp: FC<MockProps> = ({ children }) => (
  <Suspense>
    <RouterContext.Provider value={mockRouter}>
      <SessionProvider session={mockSession}>{children}</SessionProvider>
    </RouterContext.Provider>
  </Suspense>
)

const mockRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockApp, ...options })

export * from '@testing-library/react'
export { mockRender as render, mockRouter as router, mockSession as session }
