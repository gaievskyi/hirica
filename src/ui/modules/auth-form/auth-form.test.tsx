import { render } from "../../../utils/render";
import { AuthForm } from "./auth-form";

describe("Sign up form", () => {
  it("renders correctly", () => {
    const { container } = render(<AuthForm action={() => null} />);
    expect(container).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
