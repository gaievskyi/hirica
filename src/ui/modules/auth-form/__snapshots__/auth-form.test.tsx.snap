// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up form renders correctly 1`] = `
<div>
  <form
    class="flex flex-col gap-10"
  >
    <a
      class="text-2xl font-semibold"
      href="/"
    >
      Hirica â€¢ Hire and find job anonymously
    </a>
    <div
      class="flex flex-col gap-2"
    >
      <label
        for="email"
      >
        E-mail
      </label>
      <input
        class="rounded-full border-2 border-black p-4"
        name="email"
        placeholder="john.doe@email.com"
        required=""
        type="email"
      />
    </div>
    <div
      class="flex items-center justify-center gap-3"
    >
      <button
        class="rounded-full bg-black p-4 text-white hover:bg-black/80 "
        type="submit"
      >
        Next
      </button>
      <span>
        or continue with
      </span>
      <button
        class="rounded-full bg-black p-4 text-white hover:bg-black/80"
        type="submit"
      >
        Web 3.0
      </button>
      <button
        class="rounded-full bg-black p-4 text-white hover:bg-black/80"
        type="submit"
      >
        LinkedIn
      </button>
      <button
        class="rounded-full bg-black p-4 text-white hover:bg-black/80"
        type="submit"
      >
        GitHub
      </button>
    </div>
  </form>
</div>
`;
